MapboxNavigation/NavigationMapView.swift

```diff
@@ -1350,3 +1357,114 @@ extension NavigationMapView {
     }
 }
 
+// MARK: - Building Extrusion Highlights
+
+extension NavigationMapView {
+    public func highlightBuildingExtrusion(for coordinate: CLLocationCoordinate2D) {
+        if let buildingId = buildingId(coordinate: coordinate) {
+            extrudeBuildings([(buildingId, buildingExtrusionColor)], extrudeAll: true)
+        } else {
+            unhighlightBuildingExtrusions()
+        }
+    }
+    
+    public func unhighlightBuildingExtrusions() {
+        extrudeBuildings([(0, buildingExtrusionColor)], extrudeAll: true)
+    }
+    
+    private func addStyleLayersIfNecessary() {
+        if let buildingsSource = style?.source(withIdentifier: "composite") {
```

Comment:
It’s possible for the style to incorporate the Mapbox Streets source explicitly without using source compositing. In that case, the source identifier could be something arbitrary. The current road name label feature dynamically finds the Mapbox Streets source, then adds it only if it isn’t already part of the style:

https://github.com/mapbox/mapbox-navigation-ios/blob/3f361bb8735bf2671dc302484e75feb964976a89/MapboxNavigation/RouteMapViewController.swift#L645-L656